### ISSUE TITLE
Conflicting type alias is reported as conflicting submodule

### ISSUE THREAD

**Issue by bvssvni** [rust-lang/rust#24081]
Playpen http://is.gd/YTFOov

```Rust
use std::ops::Add;

type Add = Add;

fn main() {
}
```

```
<anon>:1:5: 1:18 error: import `Add` conflicts with existing submodule [E0256]
<anon>:1 use std::ops::Add;
             ^~~~~~~~~~~~~
```
